src/VecSim/utils/vecsim_stl.h

```diff
@@ -8,13 +8,21 @@
 
 namespace vecsim_stl {
 
-template <typename T>
-using vector = std::vector<T, VecsimSTLAllocator<T>>;
-
 template <typename K, typename V>
 using unordered_map = std::unordered_map<K, V, std::hash<K>, std::equal_to<K>,
                                          VecsimSTLAllocator<std::pair<const K, V>>>;
 
+template <typename T>
+class vector : public VecsimBaseObject, public std::vector<T, VecsimSTLAllocator<T>> {
+public:
+    explicit vector(const std::shared_ptr<VecSimAllocator> &alloc)
+        : VecsimBaseObject(alloc), std::vector<T, VecsimSTLAllocator<T>>(alloc) {}
+    explicit vector(size_t cap, const std::shared_ptr<VecSimAllocator> &alloc)
+        : VecsimBaseObject(alloc), std::vector<T, VecsimSTLAllocator<T>>(cap, alloc) {}
+    explicit vector(size_t cap, T val, const std::shared_ptr<VecSimAllocator> &alloc)
+        : VecsimBaseObject(alloc), std::vector<T, VecsimSTLAllocator<T>>(cap, val, alloc) {}
+};
+
```

Comment:
why were the explicit constructors needed? I recall that we used them before

Reply:
We need then now, since our vector class is now a different class than `std::vector` (it inherits from it...)